app/Http/Controllers/SettingController.php

```diff
@@ -5,33 +5,34 @@
 use App\Http\Requests\Setting\GetSettingRequest;
 use App\Http\Requests\Setting\SearchSettingRequest;
 use App\Http\Requests\Setting\UpdateSettingRequest;
+use App\Http\Resources\Setting\SearchResource;
+use App\Http\Resources\Setting\SettingResource;
+use App\Http\Resources\Setting\UpdateResource;
 use App\Services\SettingService;
-use Illuminate\Http\JsonResponse;
-use Symfony\Component\HttpFoundation\Response;
 
 class SettingController extends Controller
 {
-    public function get(GetSettingRequest $request, SettingService $service, string $key): JsonResponse
+    public function get(GetSettingRequest $request, SettingService $service, string $key): SettingResource
     {
         $result = $service->findBy('name', $key);
 
-        return response()->json($result);
+        return SettingResource::make($result);
     }
 
-    public function update(UpdateSettingRequest $request, SettingService $service, string $key): Response
+    public function update(UpdateSettingRequest $request, SettingService $service, string $key): UpdateResource
     {
         $service->update(
             ['name' => $key],
             ['value' => $request->all()]
         );
 
-        return response('', Response::HTTP_NO_CONTENT);
+        return UpdateResource::make([]);
     }
 
-    public function search(SearchSettingRequest $request, SettingService $service): JsonResponse
+    public function search(SearchSettingRequest $request, SettingService $service): SearchResource
     {
         $result = $service->search($request->onlyValidated());
 
-        return response()->json($result);
+        return SearchResource::make($result);
```

Comment:
we need to use [resources collection](https://laravel.com/docs/10.x/eloquent-resources#resource-collections) instead